File: backend/models/StepData.js
const mongoose = require('mongoose');

const stepSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  date: { type: Date, required: true },
  steps: { type: Number, required: true },
});

module.exports = mongoose.model('StepData', stepSchema);

File: frontend/lib/models/step_data.dart
class StepData {
  final String id;
  final int steps;
  final DateTime date;

  StepData({required this.id, required this.steps, required this.date});

  factory StepData.fromJson(Map<String, dynamic> json) {
    return StepData(
      id: json['_id'],
      steps: json['steps'],
      date: DateTime.parse(json['date']),
    );
  }
}